modules/repository-url/src/main/java/org/elasticsearch/common/blobstore/url/URLBlobContainer.java

```diff
@@ -107,13 +108,18 @@ public InputStream readBlob(String name) throws IOException {
         try {
             return new BufferedInputStream(getInputStream(new URL(path, name)), blobStore.bufferSizeInBytes());
         } catch (FileNotFoundException fnfe) {
-            throw new NoSuchFileException("[" + name + "] blob not found");
+            throw new NoSuchFileException("blob object [" + name + "] not found");
         }
     }
 
     @Override
     public InputStream readBlob(String blobName, long position, long length) throws IOException {
-        throw new UnsupportedOperationException();
+        final InputStream inputStream = getInputStream(new URL(path, blobName));
```

Comment:
I would prefer to keep throwing `UnsupportedOperationException` here, as we should not support something with horrible performance characteristics.

In a separate PR, I think we should also make sure that users are using "fs" repository types when accessing shared filesystems instead of URL repositories (by changing docs).

In yet a separate PR, we can remove support for "file" in URL repositories in ES 8.0.

Reply:
👍